viewing_party/party.py

```diff
@@ -1,23 +1,134 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    movie = {"title": title, "genre": genre, "rating": rating}
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    user_watchlist = user_data["watchlist"]
+    watching_index = None
+
+    for index, movie in enumerate(user_watchlist):
+        if movie["title"] == movie_title:
+            watching_index = index
+            break
+
+    if watching_index is not None:
+        add_to_watched(user_data, user_watchlist[watching_index])
+        user_watchlist.pop(watching_index)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    watched_list = user_data["watched"]
+    rating_sum = 0
+    number_of_ratings = len(watched_list)
+    if number_of_ratings == 0:
+        return 0
 
+    for movie in watched_list:
+        rating_sum += movie["rating"]
+
+    return rating_sum/number_of_ratings
+
+def get_most_watched_genre(user_data):
+    watched_list = user_data["watched"]
+    if not watched_list:
+        return None
+    genre_counter = {}
+    for movie in watched_list:
+        genre = movie["genre"]
+        if genre in genre_counter.keys():
+            genre_counter[genre] += 1
+        else:
+            genre_counter[genre] = 1
+    
+    most_watched_genre = max(genre_counter, key=genre_counter.get)
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+# ----------------------------------------- 
+def get_unique_watched(user_data):
+    friends = user_data["friends"]
+    user_movie_list = user_data["watched"]
+    unique_movie_names = set()
+    for movie in user_movie_list:
+        unique_movie_names.add(movie["title"])

```

Comment:
Very clever to only put the title in the `set`! It wasn't an issue in this project, but can you think of what might happen here if there were multiple different movies with the same title? (e.g. there are multiple movies all named "Hamlet")